lib/math/math.go

```diff
@@ -162,6 +162,37 @@ func log(thread *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwar
 	return starlark.Float(math.Log(float64(x)) / math.Log(float64(base))), nil
 }
 
+// ceil wraps the Ceil function as a Starlark built-in that
+// returns the ceiling as an Integral.
+func ceil(
+	thread *starlark.Thread,
+	_ *starlark.Builtin,
+	args starlark.Tuple,
+	kwargs []starlark.Tuple,
+) (starlark.Value, error) {
+	var x floatOrInt
+
+	if err := starlark.UnpackPositionalArgs(
+		"ceil", args, kwargs, 1, &x,
+	); err != nil {
+		return nil, err
+	}
+
+	result := math.Ceil(float64(x))
+
+	// Python's math.ceil raises an error for inf and nan.
```

Comment:
You don't need this comment: it's obvious from the need to return an int that these inputs are invalid.

Also, combine the checks and errors, and report the supplied value, not the intermediate `result`:
```
if !isInf... || isNan... {
   return nil, fmt.Errorf("ceil: invalid argument %v", x)
}
```
But if you use NumberToInt, you don't need this check.

Reply:
Thanks for pointing that out. `NumberToInt` is a great abstraction and carries the same error messages.